google/docs/credential/tool.gpt

```diff
@@ -0,0 +1,10 @@
+Name: Google Docs OAuth Credential
+Share Credential: ../../../oauth2 as google.docs
```

Comment:
Are we going back to using `Share Credential`? We stopped using it before because it makes the credentials available to all tools within the gptscript run, and it can cause potential collision issues.

Reply:
I've been following the pattern that @ibuildthecloud set in his original [oauth tools PR](https://github.com/otto8-ai/tools/pull/83/files#diff-82a77e5893de9064e559bd275770a703f9f0b8b1a21c40256400809332f12604), but I think it's a good question to ask.

The existing tools all use this, so I'm going to merge as is. If we determine we don't want to use `Share Credential` anymore, I'll follow up with a PR that updates the credentials for all our tools.